client/src/components/tests/CKANRegistryDisplay-test.js

```diff
@@ -9,49 +13,190 @@ import { Redirect } from 'react-router-dom';
         { disableLifecycleMethods: true }
       );
 
-      const result = wrapper.instance().getVisibleFields();
-      expect(result).not.toContain('foo');
-      expect(result).toContain('bar');
-      expect(result).toContain('baz');
+      mockLoadData(wrapper);
+
+      wrapper.setState({
+        currentPage: 4,
+        recordCount: 1321,
+      });
+
+      const griddleProps = wrapper.instance().getGriddleProps();
+
+      expect(griddleProps.pageProperties).toMatchObject({
+        currentPage: 4,
+        recordCount: 1321,
+        pageSize: 20,
+      });
     });
-  });
 
-  describe('handleGetPage()', () => {
-    it('sets the page number to the state', () => {
+    it('sets sort iff set in state', () => {
       const wrapper = shallow(
-        <CKANRegistryDisplay />,
+        <CKANRegistryDisplay
+          location={{ search: '' }}
+          fields={testFields}
+        />,
         { disableLifecycleMethods: true }
       );
 
-      // Mock the `loadData` method on the component as enzyme will not prevent lifecycle events for
-      // setState
-      wrapper.instance().loadData = jest.fn();
-      wrapper.update();
+      mockLoadData(wrapper);
 
-      expect(wrapper.state().currentPage).toBe(1);
+      let griddleProps = wrapper.instance().getGriddleProps();
 
-      wrapper.instance().handleGetPage(123);
-      expect(wrapper.state().currentPage).toBe(123);
+      expect(griddleProps.sortProperties).toBeFalsy();
+
+      wrapper.setState({
+        sort: { sortField: 'baz', sortAscending: true }
+      });
+
+      griddleProps = wrapper.instance().getGriddleProps();
+
+      expect(griddleProps.sortProperties[0]).toMatchObject({
+        id: 'Baz',
+        sortAscending: true,
+      });
+    });
+  });
+
+  describe('getStateFromLocation()', () => {
+    it('parses the current page from URL search parameters', () => {
+      let wrapper = shallow(
+        <CKANRegistryDisplay
+          location={{ search: '?page=45' }}
+        />,
+        { disableLifecycleMethods: true }
+      );
+
+      let stateDefaults = wrapper.instance().getStateFromLocation();
+      expect(stateDefaults.currentPage).toBe(45);
+
+      wrapper = shallow(
+        <CKANRegistryDisplay
+          location={{ search: '?foo=45' }}
+          urlKeys={{ page: 'foo' }}
+        />,
+        { disableLifecycleMethods: true }
+      );
+
+      stateDefaults = wrapper.instance().getStateFromLocation();
+      expect(stateDefaults.currentPage).toBe(45);
+    });
+
+    it('parses the sort attributes from URL search parameters', () => {
+      let wrapper = shallow(
+        <CKANRegistryDisplay
+          location={{ search: '?sort=Foo&sortdirection=ASC' }}
+        />,
+        { disableLifecycleMethods: true }
+      );
+
+      let stateDefaults = wrapper.instance().getStateFromLocation();
+      expect(stateDefaults.sort).toMatchObject({
+        sortField: 'Foo',
+        sortAscending: true,
+      });
+
+      wrapper = shallow(
```

Comment:
This could probably be a separate `it()` statement

Reply:
The only reason I didn't is that I'm testing the same thing, just with different parameters. The easiest way to trigger the URL handler is to just mount the component. If you feel strongly I'll change. Do you have a suggestion for how I label the `it` statements?

Reply:
Not strongly. Perhaps something like "it gives priority to location params over URL keys when provided" but I don't really know the context well enough. I'm happy for you to ignore this suggestion, it's super minor.

Reply:
Ok I did split it in retrospect. The second assertion was checking that `urlKeys` has an effect.